pkg/gameservers/controller.go

```diff
@@ -316,6 +320,44 @@ func (ext *Extensions) creationValidationHandler(review admissionv1.AdmissionRev
 	return review, nil
 }
 
+// creationMutationHandlerPod that mutates a GameServer pod when it is created
+// Should only be called on gameserver pod create operations.
+func (ext *Extensions) creationMutationHandlerPod(review admissionv1.AdmissionReview) (admissionv1.AdmissionReview, error) {
+	logger := runtime.NewLoggerWithSource("gke")
+	obj := review.Request.Object
+	pod := &corev1.Pod{}
+	err := json.Unmarshal(obj.Raw, pod)
+	if err != nil {
+		// If the JSON is invalid during mutation, fall through to validation. This allows OpenAPI schema validation
+		// to proceed, resulting in a more user friendly error message.
+		return review, nil
+	}
+
+	logger.WithField("review", review).Debug("creationMutationHandlerPod")
+	pod.Spec.Containers[1].Ports[0].ContainerPort = pod.Spec.Containers[1].Ports[0].HostPort
```

Comment:
Put a big TODO in here basically saying we need to deal with the case where there are multiple, mixed type ports. I think this is ok _for now_ but we'll need to deal with it before we enable the feature gate.